ui/app/components/job-status/panel/deploying.js

```diff
@@ -79,34 +86,91 @@ export default class JobStatusPanelDeployingComponent extends Component {
-        type.label
-      );
-      if (availableSlotsToFill > 0) {
-        blocks[type.label] = Array(
-          Math.min(availableSlotsToFill, jobAllocsOfType.length)
-        )
-          .fill()
-          .map((_, i) => {
-            return jobAllocsOfType[i];
-          });
-        availableSlotsToFill -= blocks[type.label].length;
-      } else {
-        blocks[type.label] = [];
-      }
-      return blocks;
+
+    let allocationCategories = this.allocTypes.reduce((categories, type) => {
+      categories[type.label] = {
+        healthy: { canary: [], nonCanary: [] },
+        unhealthy: { canary: [], nonCanary: [] },
+      };
+      return categories;
     }, {});
+
+    for (let alloc of allocationsOfDeploymentVersion) {
+      if (availableSlotsToFill <= 0) {
+        break;
+      }
+      let status = alloc.clientStatus;
+      let health = alloc.isHealthy ? 'healthy' : 'unhealthy';
+      let canary = alloc.isCanary ? 'canary' : 'nonCanary';
+
+      if (allocationCategories[status]) {
+        allocationCategories[status][health][canary].push(alloc);
+        availableSlotsToFill--;
+      }
+    }
+
+    // Fill unplaced slots if availableSlotsToFill > 0
     if (availableSlotsToFill > 0) {
-      allocationsOfShowableType['unplaced'] = Array(availableSlotsToFill)
+      allocationCategories['unplaced'] = {
+        healthy: { canary: [], nonCanary: [] },
+        unhealthy: { canary: [], nonCanary: [] },
+      };
+      allocationCategories['unplaced']['healthy']['nonCanary'] = Array(
+        availableSlotsToFill
+      )
         .fill()
         .map(() => {
           return { clientStatus: 'unplaced' };
         });
     }
-    return allocationsOfShowableType;
+
+    return allocationCategories;
+  }
+
+  get newRunningHealthyAllocBlocks() {
+    return [
+      ...this.newVersionAllocBlocks['running']['healthy']['canary'],
+      ...this.newVersionAllocBlocks['running']['healthy']['nonCanary'],
+    ];
+  }
+
+  // #region legend
+  get newAllocsByStatus() {
+    return Object.entries(this.newVersionAllocBlocks).reduce(
+      (counts, [status, healthStatusObj]) => {
+        counts[status] = Object.values(healthStatusObj)
+          .flatMap((canaryStatusObj) => Object.values(canaryStatusObj))
+          .flatMap((canaryStatusArray) => canaryStatusArray).length;
+        return counts;
+      },
+      {}
+    );
+  }
+
+  get newAllocsByCanary() {
+    return Object.values(this.newVersionAllocBlocks)
+      .flatMap((healthStatusObj) => Object.values(healthStatusObj))
+      .flatMap((canaryStatusObj) => Object.entries(canaryStatusObj))
+      .reduce((counts, [canaryStatus, items]) => {
+        counts[canaryStatus] = (counts[canaryStatus] || 0) + items.length;
+        return counts;
+      }, {});
+  }
+
+  get newAllocsByHealth() {
+    return {
+      healthy: this.newRunningHealthyAllocBlocks.length,
+      'health unknown':
+        this.totalAllocs - this.newRunningHealthyAllocBlocks.length,
+    };
+  }
```

Comment:
`thought`:  I think it might be time to really consider what `allocBlock` is doing and encapsulate that behavior into its own class. As we get deeper into computing derived state, I imagine we'll run into reactivity issues where we're inadvertently showing stale data. And that's already one of the challenges and reasons why we're considering long-polling the allocations associated to a specific deployment. We do this already for the `AllocationStatsTracker`.

This would really come in handy if we ever redesigned the Job Overview Page because we'd decouple the computation of the logic out the component. Or if we wanted to quickly follow on the feature and expand this into another view like this:

![image](https://user-images.githubusercontent.com/41024828/230183051-9f26bd87-6b30-43f9-bb87-3f2fb26a12fc.png)